plugins/imu/include/yarp/dev/IMUDriver.h

```diff
@@ -55,24 +60,188 @@ class yarp::dev::GazeboYarpIMUDriver:
+     * @param[in] sens_index sensor index (must be 0 in the case BoschIMU)
+     * @return MAS_OK/MAS_ERROR if status ok/failure
+     */
+    yarp::dev::MAS_status getThreeAxisMagnetometerStatus(size_t sens_index) const override;
+
+    /**
+     * Get the name of three axis magnetometer
+     * @param[in] sens_index sensor index (must be 0 in the case BoschIMU)
+     * @param[out] name name of the sensor
+     * @return true/false success/failure
+     */
+    bool getThreeAxisMagnetometerName(size_t sens_index, std::string &name) const override;
+
+    /**
+     * Get the name of the frame in which three axis magnetometer measurements are expressed
+     * @param[in] sens_index sensor index (must be 0 in the case BoschIMU)
+     * @param[out] frameName name of the sensor frame
+     * @return true/false success/failure
+     */
+    bool getThreeAxisMagnetometerFrameName(size_t sens_index, std::string &frameName) const override;
+
+    /**
+     * Get three axis magnetometer measurements
+     * @param[in] sens_index sensor index (must be 0 in the case BoschIMU)
+     * @param[out] out 3D magnetometer measurement
+     * @param[out] timestamp timestamp of measurement
+     * @return true/false success/failure
+     */
+    bool getThreeAxisMagnetometerMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const override;
+
+    /* IOrientationSensors methods */
+    /**
+     * Get the  number of orientation sensors in the device
+     * @return 1
+     */
+    size_t getNrOfOrientationSensors() const override;
+
+    /**
+     * Get the status of orientation sensor
+     * @param[in] sens_index sensor index (must be 0 in the case BoschIMU)
+     * @return MAS_OK/MAS_ERROR if status ok/failure
+     */
+    yarp::dev::MAS_status getOrientationSensorStatus(size_t sens_index) const override;
+
+    /**
+     * Get the name of orientation sensor
+     * @param[in] sens_index sensor index (must be 0 in the case BoschIMU)
+     * @param[out] name name of the sensor
+     * @return true/false success/failure
+     */
+    bool getOrientationSensorName(size_t sens_index, std::string &name) const override;
+
+    /**
+     * Get the name of the frame in which orientation sensor measurements are expressed
+     * @param[in] sens_index sensor index (must be 0 in the case BoschIMU)
+     * @param[out] frameName name of the sensor frame
+     * @return true/false success/failure
+     */
+    bool getOrientationSensorFrameName(size_t sens_index, std::string &frameName) const override;
+
+    /**
+     * Get orientation sensor measurements
+     * @param[in] sens_index sensor index (must be 0 in the case BoschIMU)
+     * @param[out] out RPY Euler angles in deg
+     * @param[out] timestamp timestamp of measurement
+     * @return true/false success/failure
+     */
+    bool getOrientationSensorMeasureAsRollPitchYaw(size_t sens_index, yarp::sig::Vector& rpy, double& timestamp) const override;
```

Comment:
I would just drop all the documentation of this methods. The methods are already documented in YARP, and copy pasting the docs just make them prone to be outdated if the YARP docs are updated.